### PR TITLE
Lesson 4.1: Performance TypeScript Testing

### PR DESCRIPTION
**Pull request by verafes**

### CODE REVIEW

File: src/components/TodoContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                    <AddTodoForm onAddTodo={addTodo} todoList={todoList}/>
+                    <TodoList todoList={todoList} onRemoveTodo={removeTodo}/>
+                </>
+            )}
+        </>
+    )
+}
+
+TodoContainer.propTypes = {
+    tableName: PropTypes.string.isRequired,
```
Comment (AnnaPestova1) [inline src/components/TodoContainer.jsx:127]: just interesting: why only in this component you have a flag: isRequires? Does it mean in other components the props are optional?

File: src/components/TodoList.jsx [hunk 2]
```diff
@@ -1,5 +1,7 @@
 import TodoListItem from './TodoListItem.jsx';
-import style from './TodoListItem.module.css';
+import PropTypes from "prop-types";
+import TodoContainer from "./TodoContainer.jsx";
```
Comment (AnnaPestova1) [inline src/components/TodoList.jsx]: do you need this import here?

**Review by AnnaPestova1 — APPROVED** [review]
Hi Vera! Great work! Have small comments that do not affect your work completion.

File: src/components/TodoContainer.jsx (hunk 1 above)
Comment (verafes) [reply src/components/TodoContainer.jsx:127]: When building components for applications that interact with APIs, using .isRequired for props is indeed super important.

File: src/components/TodoList.jsx (hunk 2 above)
Comment (verafes) [reply src/components/TodoList.jsx]: Removed, thank you!

**Pull request merged**